lib/shelley/bench/Restore.hs

```diff
@@ -9,6 +9,7 @@
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NoMonoLocalBinds #-}
```

Comment:
ok, will need to read about it :sweat_smile: At this moment do not understand why it is used

Reply:
So we have the main `bench_results` function with a bunch of arguments, type parameters and constraints.

I wanted to define more descriptive helpers where I apply most of the boring arguments, e.g.:
```
    let benchRestoreRndWithOwnership permillProxy = bench_restoration
            networkProxy
            (trMessageText tr)
            socketFile
            np
            vData
            [walletRnd (showPercent permillProxy <> "-percent-rnd")
                $ mkRndAnyState' permillProxy networkProxy]
            True -- Write progress to .timelog file
            (unsafeMkPercentage 1)
            benchmarksRnd

```

But the per-mille ownership is a type parameter, and Haskell will fix it when inferring the type of the let binding. 
```
benchRestoreRndWithOwnership (Proxy @0) -- compiles
benchRestoreRndWithOwnership (Proxy @1) -- Expected type: Proxy 0; Actual type: Proxy 1
```

Moving it to a where-clause, and explicitly specifying `forall (p :: Nat)` didn't seem like a good solution because the full type signature is really big and cumbersome, and I wouldn't have access to the locally bound `vData`, `np`, `networkProxy`.

With `NoMonoLocalBinds` it just works.

Looking [here](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#let-generalisation) it seems that `MonoLocalBinds` is off by default, but enabled automatically by `GADTs`. Didn't know that.

> The extension MonoLocalBinds is implied by TypeFamilies and GADTs. You can switch it off again with NoMonoLocalBinds but type inference becomes less predictable if you do so. (Read the papers!)

Reply:
Thanks!